File: src/types/model/merchant.ts
export interface Merchant {
  id: number;
  name: string;
  api_key: string;
  user_id: number;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface MerchantTrashed extends Merchant {
  deleted_at: string;
}

export interface MerchantTransaction {
  id: number;
  card_number: string;
  amount: number;
  payment_method: string;
  merchant_id: number;
  merchant_name: string;
  transaction_time: string;
  created_at: string;
  updated_at: string;
  deleted_at: string;
}

export interface MerchantMonthlyPaymentMethod {
  month: string;
  payment_method: string;
  total_amount: number;
}

export interface MerchantYearlyPaymentMethod {
  year: string;
  payment_method: string;
  total_amount: number;
}

export interface MerchantMonthlyAmount {
  month: string;
  total_amount: number;
}

export interface MerchantYearlyAmount {
  year: string;
  total_amount: number;
}

export interface MerchantMonthlyTotalAmount {
  month: string;
  year: string;
  total_amount: number;
}

export interface MerchantYearlyTotalAmount {
  year: string;
  total_amount: number;
}

File: src-tauri/src/model/merchant.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct MerchantResponse {
    pub id: i32,
    pub name: String,
    pub user_id: i32,
    pub api_key: String,
    pub status: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct MerchantResponseDeleteAt {
    pub id: i32,
    pub name: String,
    pub user_id: i32,
    pub api_key: String,
    pub status: String,
    pub created_at: String,
    pub updated_at: String,
    pub deleted_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct MerchantTransactionResponse {
    pub id: i32,
    pub card_number: String,
    pub amount: i32,
    pub payment_method: String,
    pub merchant_id: i32,
    pub merchant_name: String,
    pub transaction_time: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Serialize, Deserialize)]
pub struct MerchantResponseMonthlyPaymentMethod {
    pub month: String,
    pub payment_method: String,
    pub total_amount: i32,
}

#[derive(Serialize, Deserialize)]
pub struct MerchantResponseYearlyPaymentMethod {
    pub year: String,
    pub payment_method: String,
    pub total_amount: i32,
}

#[derive(Serialize, Deserialize)]
pub struct MerchantResponseMonthlyAmount {
    pub month: String,
    pub total_amount: i32,
}

#[derive(Serialize, Deserialize)]
pub struct MerchantResponseYearlyAmount {
    pub year: String,
    pub total_amount: i32,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct MerchantMonthlyTotalAmount {
    pub month: String,
    pub year: String,
    pub total_amount: f64,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct MerchantYearlyTotalAmount {
    pub year: String,
    pub total_amount: f64,
}
